### ISSUE TITLE
Remove *QueryFilter workaround

### ISSUE THREAD

**Issue by morremeyer** [envelope-zero/backend#223]
Currently, there is a `controllers.*QueryFilter` struct for each resource type.

This is needed to bind the e.g. BudgetID to a string and then marshal it into a UUID.

Once https://github.com/gin-gonic/gin/pull/3045 is merged, this struct can be removed and the tags merged into the tags of the `controllers.*Editable` struct of the respective resource type, adding marshalling as by the documentation that the PR adds.

**Comment by morremeyer** [comment]
This is unblocked with #1035

**Comment by morremeyer** [comment]
Since the query filters have been extended a lot (e.g. to support filtering of envelopes by budget ID), we want to keep them.

What should be removed is the explicit UUID parsing.
